magento/magento2
# Add logic check only import css from enabled modules

### Description (*)

This PR based on workarounds @vasilii-b for resolve this remain issue. Default magento import all css from all modules in (app/etc/config.php) no matter it was enabled or not. 
See file less in var/view_preprocessed/pub/static/frontend/Vendor/theme/locale/css
For normal site with small customise module css size is not problem but imagine store have a lot of modules and each module have custom styles. Css output size will be huge and delay browser parse css 

when see file styles-m.less or styles-l.less, you can see this line

```
//@magento_import 'source/_module.less'; // Theme modules
//@magento_import 'source/_widgets.less'; // Theme widgets
```

It's not actually comment code. This is syntax by magento allow import external modules and widgets styles across in all modules and themes

Current exists problem
If we disabled module but we add styles inside theme -> these styles in theme also added to final output instead expect not available in final css.

Benefits/Expects
assume If module disabled so that styles listed below should not be include in output styles
```
@import '../Vendor_Module/css/source/_extend.less'; //Should not include
@import '../Vendor_Module/css/source/_module.less'; //Should not include
@import '../Vendor_Module/css/source/_widgets.less'; //Should not include
```
disabled module is deprecated but some case if we need or know exactly problem come from some module. We can disabled it and also make sure styles come from theme of that module will not parse at final output css.
The benefits => reduce size outputs css file

Some downside/edge-cases

Case 1 (Rarely): If we add styles of disabled module to enabled module. We can't exclude it. I'm not sure in real-world we may got this case (look like it's related with manager code/quality)
Example
Magento_Review -> disabled

in theme luma theme-frontend-luma/Magento_Catalog/web/css/source/_extend.less

```less
@import 'Magento_Review::css/custom.less';
```
theme-frontend-luma/Magento_Review/web/css/custom.less <= styles in here still loaded. Only styles from files, _module.less, widget, _extend will able wipe out

Known limitations
Some variables added/defined in module less. But some other files such as (_extends.less) reference/use/call to it. One case will false if we disabled module
Solutions: Need refactor files to loose tight-coupling


### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#24666

### Manual testing scenarios (*)

1. Magento 2.3.4+
2. blank or luma theme
3. Disable a couple module use less styles: Ex: Magento_Review, Magento_LoginAsCustomerFrontendUi, PageBuilder (note backend function will not work, but frontend styles will expected not available)
4. Add  inside to app/etc/env.php or app/etc/config.php
  ```php
  'static_content_only_enabled_modules' => true,
  ```

5. Run setup upgrade and deploy static content and clear cache
6. Find class belong modules disabled should not available at final css output files(styles-m.css,styles-l.css)

### Questions or comments


### Contribution checklist (*)
